### PR TITLE
feat: [IOPAE-573] add backoffice infra resources

### PR DESCRIPTION
**Pull request by rodoviti**
Add backoffice infra resources and refactor-move terraform variables

#### List of Changes

- add app service
- refactor: move all variables inside `99_variables.tf` file
- refactor: move all "io common" variables into a "single object" variable
- refactor: move all "subnets cidrs" variables into a "single object" variable
- refactor: move all "subnets" definitions inside `network.tf` file

#### Motivation and Context

To "serve" the new backoffice

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: infra/src/00_network.tf [hunk 1]
```diff
@@ -2,45 +2,58 @@ data "azurerm_subnet" "github_runner_subnet" {
...
   name                = "privatelink.postgres.database.azure.com"
-  resource_group_name = var.vnet_common_rg
+  resource_group_name = var.io_common.resource_group_name
 }
 
+
+#data "azurerm_private_dns_zone" "privatelink_azurewebsites_net" { # FIXME: is it required ?
+#  name                = "privatelink.azurewebsites.net"
+#  resource_group_name = var.io_common.resource_group_name
+#}
```
Comment (pasqualedevita) [inline infra/src/00_network.tf:36]: only if app service/function is configured with private endpoints

File: infra/src/backoffice_webapp.tf [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  app_settings = local.backoffice_app_settings
+
+  always_on        = true
+  vnet_integration = true
+
+  subnet_id = module.backoffice_app_snet.id
+
+  allowed_subnets = [
+    data.azurerm_subnet.appgateway_snet.id,
+    # data.azurerm_subnet.apim_v2_snet[0].id # FIXME: is it required? (apim should not call the app service, but app service should query APIM through its SDK [so I think they will go through his public endpoint])
```
Comment (pasqualedevita) [inline infra/src/backoffice_webapp.tf:40]: not required because the flow is app gateway -> app service.

File: infra/src/backoffice_webapp.tf [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  ]
+
+  tags = var.tags
+}
+
+#resource "azurerm_private_endpoint" "backoffice_app" { # FIXME: is it required? (backoffice app service should be called only through appgateway, so we have-to/can disable public network access to eliminating public exposure)
```
Comment (pasqualedevita) [inline infra/src/backoffice_webapp.tf:46]: not required

File: infra/src/env/prod/terraform.tfvars [hunk 4]
```diff
@@ -68,4 +75,23 @@ legacy_service_watcher_max_items_per_invocation = 10
 userid_cms_to_legacy_sync_inclusion_list         = "*"
 userid_legacy_to_cms_sync_inclusion_list         = "*"
 userid_request_review_legacy_inclusion_list      = "*"
-userid_automatic_service_approval_inclusion_list = ""
\ No newline at end of file
+userid_automatic_service_approval_inclusion_list = ""
+
+# Backoffice Configurations
+backoffice_app = {
+  sku_name = "S1" # FIXME: use "P1v3" before "production launch"
```
Comment (pasqualedevita) [inline infra/src/env/prod/terraform.tfvars:82]: ok

File: infra/src/network.tf [hunk 5]
```diff
@@ -35,3 +26,53 @@ module "app_snet" {
...
+#
+
+module "backoffice_app_snet" {
+  source               = "git::https://github.com/pagopa/terraform-azurerm-v3.git//subnet?ref=v6.20.2"
+  name                 = format("%s-%s-backoffice-snet", local.project, local.application_basename)
+  resource_group_name  = var.io_common.resource_group_name
+  virtual_network_name = var.io_common.vnet_name
+  address_prefixes     = var.subnets_cidrs.backoffice
+
+  private_endpoint_network_policies_enabled = false # FIXME: is it correct?
```
Comment (pasqualedevita) [inline infra/src/network.tf:65]: I don't remember try false

File: infra/src/key_vault.tf [hunk 6]
```diff
@@ -79,7 +79,7 @@ resource "azurerm_key_vault_access_policy" "github_action_cd" {
   key_permissions         = []
 }
 
-resource "azurerm_key_vault_secret" "webapp_fn_app_key" {
+resource "azurerm_key_vault_secret" "webapp_fn_app_key" { # FIXME: remove
```
[the remark this answers is not in the document]
Comment (rodoviti) [reply infra/src/key_vault.tf:82]: @pasqualedevita i think we could remove this resource if we change [this line](https://github.com/pagopa/io-services-cms/blob/IOPAE-573--infra-backoffice/infra/src/apim_api.tf#L57) with the suggestion on the comment, does that work for you?

**Review by giuseppedipinto — APPROVED** [review]

**Pull request merged**